Synthesis failed for nodejs-dns

Hello! Autosynth couldn't regenerate nodejs-dns. :broken_heart:

Here's the output from running `synth.py`:

```
in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-20 06:07:25,522 autosynth [DEBUG] > Running: git checkout 274dd49554809834287c24b6dd324a85283f1182
Previous HEAD position was be74d3e build: do not fail builds on codecov errors (#528)
HEAD is now at 274dd49 fix: autosynth for nodejs-apiary (#569)
2020-05-20 06:07:25,534 autosynth [DEBUG] > Running: git branch -f autosynth-26
2020-05-20 06:07:25,537 autosynth [DEBUG] > Running: git checkout autosynth-26
Switched to branch 'autosynth-26'
2020-05-20 06:07:25,541 autosynth [INFO] > Running synthtool
2020-05-20 06:07:25,541 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-20 06:07:25,543 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-05-20 06:07:25,758 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-dns/synth.py.
On branch autosynth-26
nothing to commit, working tree clean
2020-05-20 06:07:25,894 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-20 06:07:26,030 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-dns
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-dns
> gts clean

version: 12
Removing build ...

> @google-cloud/dns@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-dns
> tsc -p .

[96mtest/change.ts[0m:[93m76[0m:[93m33[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m76[0m       const calledWith = change.calledWith_[0];
[7m  [0m [91m                                ~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m126[0m:[93m30[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m126[0m       const calledWith = dns.calledWith_[0];
[7m   [0m [91m                             ~~~~~~~~~~~[0m

[96mtest/zone.ts[0m:[93m169[0m:[93m31[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m169[0m       const calledWith = zone.calledWith_[0];
[7m   [0m [91m                              ~~~~~~~~~~~[0m


Found 3 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-20T13_07_35_128Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-20T13_07_35_144Z-debug.log
2020-05-20 06:07:35,156 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
2020-05-20 06:07:35,157 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-dns/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-20 06:07:35,193 autosynth [ERROR] > Synthesis failed
2020-05-20 06:07:35,193 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-20 06:07:35,200 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-05-20 06:07:35,204 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 615, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 476, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 595, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 371, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 381, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 119, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge/1d96130f-887f-4822-a400-bff75bdfb430).

Autosynth is still having trouble generating nodejs-dns. :sob:

Here's the output from running `synth.py`:

```
in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-21 05:11:53,919 autosynth [DEBUG] > Running: git checkout 7ee92820e64c0aea379781b82399d6b3f3c8655f
Previous HEAD position was be74d3e build: do not fail builds on codecov errors (#528)
HEAD is now at 7ee9282 fix: don't fail when clean fails (#574)
2020-05-21 05:11:53,932 autosynth [DEBUG] > Running: git branch -f autosynth-28
2020-05-21 05:11:53,936 autosynth [DEBUG] > Running: git checkout autosynth-28
Switched to branch 'autosynth-28'
2020-05-21 05:11:53,942 autosynth [INFO] > Running synthtool
2020-05-21 05:11:53,942 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-21 05:11:53,945 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-05-21 05:11:54,161 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-dns/synth.py.
On branch autosynth-28
nothing to commit, working tree clean
2020-05-21 05:11:54,299 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-21 05:11:54,428 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-dns
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-dns
> gts clean

version: 12
Removing build ...

> @google-cloud/dns@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-dns
> tsc -p .

[96mtest/change.ts[0m:[93m76[0m:[93m33[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m76[0m       const calledWith = change.calledWith_[0];
[7m  [0m [91m                                ~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m126[0m:[93m30[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m126[0m       const calledWith = dns.calledWith_[0];
[7m   [0m [91m                             ~~~~~~~~~~~[0m

[96mtest/zone.ts[0m:[93m169[0m:[93m31[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m169[0m       const calledWith = zone.calledWith_[0];
[7m   [0m [91m                              ~~~~~~~~~~~[0m


Found 3 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-21T12_12_03_250Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-21T12_12_03_270Z-debug.log
2020-05-21 05:12:03,285 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
2020-05-21 05:12:03,286 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-dns/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-21 05:12:03,340 autosynth [ERROR] > Synthesis failed
2020-05-21 05:12:03,340 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-21 05:12:03,348 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-05-21 05:12:03,354 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 615, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 476, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 595, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 371, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 381, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 119, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge/2d09a9e0-e49f-4d2b-b5ea-2b554f1b32ec).

Autosynth is still having trouble generating nodejs-dns. :sob:

Here's the output from running `synth.py`:

```
ound, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-22 05:59:14,554 autosynth [DEBUG] > Running: git checkout 65f8c647c0bc0c6b38211b969a2a003e271a5ef1
Previous HEAD position was be74d3e build: do not fail builds on codecov errors (#528)
HEAD is now at 65f8c64 chore(java): allow overriding minimum supported Java version in templates (#577)
2020-05-22 05:59:14,565 autosynth [DEBUG] > Running: git branch -f autosynth-29
2020-05-22 05:59:14,569 autosynth [DEBUG] > Running: git checkout autosynth-29
Switched to branch 'autosynth-29'
2020-05-22 05:59:14,574 autosynth [INFO] > Running synthtool
2020-05-22 05:59:14,574 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-22 05:59:14,576 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-05-22 05:59:14,806 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-dns/synth.py.
On branch autosynth-29
nothing to commit, working tree clean
2020-05-22 05:59:14,947 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-22 05:59:15,091 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-dns
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-dns
> gts clean

version: 12
Removing build ...

> @google-cloud/dns@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-dns
> tsc -p .

[96mtest/change.ts[0m:[93m76[0m:[93m33[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m76[0m       const calledWith = change.calledWith_[0];
[7m  [0m [91m                                ~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m126[0m:[93m30[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m126[0m       const calledWith = dns.calledWith_[0];
[7m   [0m [91m                             ~~~~~~~~~~~[0m

[96mtest/zone.ts[0m:[93m169[0m:[93m31[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m169[0m       const calledWith = zone.calledWith_[0];
[7m   [0m [91m                              ~~~~~~~~~~~[0m


Found 3 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-22T12_59_24_431Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-22T12_59_24_447Z-debug.log
2020-05-22 05:59:24,460 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
2020-05-22 05:59:24,461 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-dns/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-22 05:59:24,506 autosynth [ERROR] > Synthesis failed
2020-05-22 05:59:24,506 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-22 05:59:24,513 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-05-22 05:59:24,518 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 615, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 476, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 595, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 371, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 381, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 119, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge/8aeddfbb-7b58-4fc8-9895-2ada12afbca7).

Autosynth is still having trouble generating nodejs-dns. :sob:

Here's the output from running `synth.py`:

```
ound, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-23 12:06:31,526 autosynth [DEBUG] > Running: git checkout 65f8c647c0bc0c6b38211b969a2a003e271a5ef1
Previous HEAD position was be74d3e build: do not fail builds on codecov errors (#528)
HEAD is now at 65f8c64 chore(java): allow overriding minimum supported Java version in templates (#577)
2020-05-23 12:06:31,540 autosynth [DEBUG] > Running: git branch -f autosynth-29
2020-05-23 12:06:31,545 autosynth [DEBUG] > Running: git checkout autosynth-29
Switched to branch 'autosynth-29'
2020-05-23 12:06:31,551 autosynth [INFO] > Running synthtool
2020-05-23 12:06:31,552 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-23 12:06:31,555 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-05-23 12:06:31,804 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-dns/synth.py.
On branch autosynth-29
nothing to commit, working tree clean
2020-05-23 12:06:31,956 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-23 12:06:32,102 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-dns
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-dns
> gts clean

version: 12
Removing build ...

> @google-cloud/dns@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-dns
> tsc -p .

[96mtest/change.ts[0m:[93m76[0m:[93m33[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m76[0m       const calledWith = change.calledWith_[0];
[7m  [0m [91m                                ~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m126[0m:[93m30[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m126[0m       const calledWith = dns.calledWith_[0];
[7m   [0m [91m                             ~~~~~~~~~~~[0m

[96mtest/zone.ts[0m:[93m169[0m:[93m31[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m169[0m       const calledWith = zone.calledWith_[0];
[7m   [0m [91m                              ~~~~~~~~~~~[0m


Found 3 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-23T19_06_41_731Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-23T19_06_41_749Z-debug.log
2020-05-23 12:06:41,764 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
2020-05-23 12:06:41,766 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-dns/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-23 12:06:41,828 autosynth [ERROR] > Synthesis failed
2020-05-23 12:06:41,828 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-23 12:06:41,837 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-05-23 12:06:41,844 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 615, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 476, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 595, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 371, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 381, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 119, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge/55b41abd-49e7-468a-beba-a8683b58c54c).

Autosynth is still having trouble generating nodejs-dns. :sob:

Here's the output from running `synth.py`:

```
ound, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-24 10:02:29,491 autosynth [DEBUG] > Running: git checkout 65f8c647c0bc0c6b38211b969a2a003e271a5ef1
Previous HEAD position was be74d3e build: do not fail builds on codecov errors (#528)
HEAD is now at 65f8c64 chore(java): allow overriding minimum supported Java version in templates (#577)
2020-05-24 10:02:29,504 autosynth [DEBUG] > Running: git branch -f autosynth-29
2020-05-24 10:02:29,507 autosynth [DEBUG] > Running: git checkout autosynth-29
Switched to branch 'autosynth-29'
2020-05-24 10:02:29,512 autosynth [INFO] > Running synthtool
2020-05-24 10:02:29,512 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-24 10:02:29,514 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-05-24 10:02:29,740 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-dns/synth.py.
On branch autosynth-29
nothing to commit, working tree clean
2020-05-24 10:02:29,879 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-24 10:02:30,021 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-dns
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-dns
> gts clean

version: 12
Removing build ...

> @google-cloud/dns@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-dns
> tsc -p .

[96mtest/change.ts[0m:[93m76[0m:[93m33[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m76[0m       const calledWith = change.calledWith_[0];
[7m  [0m [91m                                ~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m126[0m:[93m30[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m126[0m       const calledWith = dns.calledWith_[0];
[7m   [0m [91m                             ~~~~~~~~~~~[0m

[96mtest/zone.ts[0m:[93m169[0m:[93m31[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m169[0m       const calledWith = zone.calledWith_[0];
[7m   [0m [91m                              ~~~~~~~~~~~[0m


Found 3 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-24T17_02_39_331Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-24T17_02_39_351Z-debug.log
2020-05-24 10:02:39,367 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
2020-05-24 10:02:39,368 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-dns/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-24 10:02:39,415 autosynth [ERROR] > Synthesis failed
2020-05-24 10:02:39,415 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-24 10:02:39,422 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-05-24 10:02:39,427 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 615, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 476, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 595, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 371, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 381, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 119, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge/1b789978-f73a-46c5-b32f-b0f5a9fa15cc).

Autosynth is still having trouble generating nodejs-dns. :sob:

Here's the output from running `synth.py`:

```
ound, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-25 08:05:39,796 autosynth [DEBUG] > Running: git checkout 65f8c647c0bc0c6b38211b969a2a003e271a5ef1
Previous HEAD position was be74d3e build: do not fail builds on codecov errors (#528)
HEAD is now at 65f8c64 chore(java): allow overriding minimum supported Java version in templates (#577)
2020-05-25 08:05:39,810 autosynth [DEBUG] > Running: git branch -f autosynth-29
2020-05-25 08:05:39,814 autosynth [DEBUG] > Running: git checkout autosynth-29
Switched to branch 'autosynth-29'
2020-05-25 08:05:39,818 autosynth [INFO] > Running synthtool
2020-05-25 08:05:39,819 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-25 08:05:39,821 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-05-25 08:05:40,042 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-dns/synth.py.
On branch autosynth-29
nothing to commit, working tree clean
2020-05-25 08:05:40,172 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-25 08:05:40,309 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-dns
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-dns
> gts clean

version: 12
Removing build ...

> @google-cloud/dns@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-dns
> tsc -p .

[96mtest/change.ts[0m:[93m76[0m:[93m33[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m76[0m       const calledWith = change.calledWith_[0];
[7m  [0m [91m                                ~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m126[0m:[93m30[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m126[0m       const calledWith = dns.calledWith_[0];
[7m   [0m [91m                             ~~~~~~~~~~~[0m

[96mtest/zone.ts[0m:[93m169[0m:[93m31[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m169[0m       const calledWith = zone.calledWith_[0];
[7m   [0m [91m                              ~~~~~~~~~~~[0m


Found 3 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-25T15_05_49_335Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-25T15_05_49_353Z-debug.log
2020-05-25 08:05:49,366 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
2020-05-25 08:05:49,367 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-dns/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-25 08:05:49,402 autosynth [ERROR] > Synthesis failed
2020-05-25 08:05:49,403 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-25 08:05:49,410 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-05-25 08:05:49,415 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 615, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 476, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 595, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 371, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 381, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 119, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge/ba51bb8e-983e-4ef5-846f-55ea9fa4e79c).

Autosynth is still having trouble generating nodejs-dns. :sob:

Here's the output from running `synth.py`:

```
ound, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-26 07:16:34,613 autosynth [DEBUG] > Running: git checkout 65f8c647c0bc0c6b38211b969a2a003e271a5ef1
Previous HEAD position was be74d3e build: do not fail builds on codecov errors (#528)
HEAD is now at 65f8c64 chore(java): allow overriding minimum supported Java version in templates (#577)
2020-05-26 07:16:34,625 autosynth [DEBUG] > Running: git branch -f autosynth-29
2020-05-26 07:16:34,628 autosynth [DEBUG] > Running: git checkout autosynth-29
Switched to branch 'autosynth-29'
2020-05-26 07:16:34,632 autosynth [INFO] > Running synthtool
2020-05-26 07:16:34,632 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-26 07:16:34,634 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-05-26 07:16:34,855 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-dns/synth.py.
On branch autosynth-29
nothing to commit, working tree clean
2020-05-26 07:16:34,991 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-26 07:16:35,130 synthtool [DEBUG] > Installing dependencies...
DEBUG:synthtool:Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-dns
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/dns@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-dns
> gts clean

version: 12
Removing build ...

> @google-cloud/dns@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-dns
> tsc -p .

[96mtest/change.ts[0m:[93m76[0m:[93m33[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m76[0m       const calledWith = change.calledWith_[0];
[7m  [0m [91m                                ~~~~~~~~~~~[0m

[96mtest/index.ts[0m:[93m126[0m:[93m30[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m126[0m       const calledWith = dns.calledWith_[0];
[7m   [0m [91m                             ~~~~~~~~~~~[0m

[96mtest/zone.ts[0m:[93m169[0m:[93m31[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m169[0m       const calledWith = zone.calledWith_[0];
[7m   [0m [91m                              ~~~~~~~~~~~[0m


Found 3 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-26T14_16_44_376Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/dns@2.0.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/dns@2.0.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-26T14_16_44_393Z-debug.log
2020-05-26 07:16:44,405 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
2020-05-26 07:16:44,406 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-dns/synth.py", line 13, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-26 07:16:44,445 autosynth [ERROR] > Synthesis failed
2020-05-26 07:16:44,446 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)
2020-05-26 07:16:44,453 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-05-26 07:16:44,458 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 615, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 476, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 595, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 371, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 381, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 119, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge/0344a2cb-00c6-4ee5-a325-b78b8a82b7b8).

Autosynth is still having trouble generating nodejs-dns. :sob:

Here's the output from running `synth.py`:

```
b'n look around, make experimental\nchanges and commit them, and you can discard any commits you make in this\nstate without impacting any branches by performing another checkout.\n\nIf you want to create a new branch to retain commits you create, you may\ndo so (now or later) by using -b with the checkout command again. Example:\n\n  git checkout -b <new-branch-name>\n\nHEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)\n2020-05-27 05:08:53,603 autosynth [DEBUG] > Running: git checkout 470789cee75ce93c41348ad6aa4c49363a80399b\nPrevious HEAD position was be74d3e build: do not fail builds on codecov errors (#528)\nHEAD is now at 470789c chore(java): move version update end tag below client dependency (#579)\n2020-05-27 05:08:53,616 autosynth [DEBUG] > Running: git branch -f autosynth-32\n2020-05-27 05:08:53,620 autosynth [DEBUG] > Running: git checkout autosynth-32\nSwitched to branch \'autosynth-32\'\n2020-05-27 05:08:53,625 autosynth [INFO] > Running synthtool\n2020-05-27 05:08:53,625 autosynth [INFO] > [\'/tmpfs/src/github/synthtool/env/bin/python3\', \'-m\', \'synthtool\', \'--metadata\', \'synth.metadata\', \'synth.py\', \'--\']\n2020-05-27 05:08:53,628 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --\n2020-05-27 05:08:53,838 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/nodejs-dns/synth.py.\nOn branch autosynth-32\nnothing to commit, working tree clean\n2020-05-27 05:08:53,970 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool\nDEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool\n.eslintignore\n.eslintrc.json\n.github/ISSUE_TEMPLATE/bug_report.md\n.github/ISSUE_TEMPLATE/feature_request.md\n.github/ISSUE_TEMPLATE/support_request.md\n.github/PULL_REQUEST_TEMPLATE.md\n.github/publish.yml\n.github/release-please.yml\n.github/workflows/ci.yaml\n.kokoro/common.cfg\n.kokoro/continuous/node10/common.cfg\n.kokoro/continuous/node10/docs.cfg\n.kokoro/continuous/node10/lint.cfg\n.kokoro/continuous/node10/samples-test.cfg\n.kokoro/continuous/node10/system-test.cfg\n.kokoro/continuous/node10/test.cfg\n.kokoro/continuous/node12/common.cfg\n.kokoro/continuous/node12/test.cfg\n.kokoro/continuous/node8/common.cfg\n.kokoro/continuous/node8/test.cfg\n.kokoro/docs.sh\n.kokoro/lint.sh\n.kokoro/presubmit/node10/common.cfg\n.kokoro/presubmit/node10/docs.cfg\n.kokoro/presubmit/node10/lint.cfg\n.kokoro/presubmit/node10/samples-test.cfg\n.kokoro/presubmit/node10/system-test.cfg\n.kokoro/presubmit/node10/test.cfg\n.kokoro/presubmit/node12/common.cfg\n.kokoro/presubmit/node12/test.cfg\n.kokoro/presubmit/node8/common.cfg\n.kokoro/presubmit/node8/test.cfg\n.kokoro/presubmit/windows/common.cfg\n.kokoro/presubmit/windows/test.cfg\n.kokoro/publish.sh\n.kokoro/release/docs.cfg\n.kokoro/release/docs.sh\n.kokoro/release/publish.cfg\n.kokoro/samples-test.sh\n.kokoro/system-test.sh\n.kokoro/test.bat\n.kokoro/test.sh\n.kokoro/trampoline.sh\n.mocharc.js\n.nycrc\n.prettierignore\n.prettierrc.js\nCODE_OF_CONDUCT.md\nCONTRIBUTING.md\nLICENSE\nREADME.md\nrenovate.json\nsamples/README.md\n2020-05-27 05:08:54,095 synthtool [DEBUG] > Installing dependencies...\nDEBUG:synthtool:Installing dependencies...\nnpm WARN npm npm does not support Node.js v12.17.0\nnpm WARN npm You should probably upgrade to a newer version of node as we\nnpm WARN npm can\'t make any promises that npm will work with this version.\nnpm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.\nnpm WARN npm You can find the latest version at https://nodejs.org/\n\n> @google-cloud/dns@2.0.0 prepare /home/kbuilder/.cache/synthtool/nodejs-dns\n> npm run compile\n\nnpm WARN npm npm does not support Node.js v12.17.0\nnpm WARN npm You should probably upgrade to a newer version of node as we\nnpm WARN npm can\'t make any promises that npm will work with this version.\nnpm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.\nnpm WARN npm You can find the latest version at https://nodejs.org/\n\n> @google-cloud/dns@2.0.0 precompile /home/kbuilder/.cache/synthtool/nodejs-dns\n> gts clean\n\nversion: 12\nRemoving build ...\n\n> @google-cloud/dns@2.0.0 compile /home/kbuilder/.cache/synthtool/nodejs-dns\n> tsc -p .\n\n\x1b[96mtest/change.ts\x1b[0m:\x1b[93m76\x1b[0m:\x1b[93m33\x1b[0m - \x1b[91merror\x1b[0m\x1b[90m TS2339: \x1b[0mProperty \'calledWith_\' does not exist on type \'ServiceObject<any>\'.\n\n\x1b[7m76\x1b[0m       const calledWith = change.calledWith_[0];\n\x1b[7m  \x1b[0m \x1b[91m                                ~~~~~~~~~~~\x1b[0m\n\n\x1b[96mtest/index.ts\x1b[0m:\x1b[93m126\x1b[0m:\x1b[93m30\x1b[0m - \x1b[91merror\x1b[0m\x1b[90m TS2339: \x1b[0mProperty \'calledWith_\' does not exist on type \'Service\'.\n\n\x1b[7m126\x1b[0m       const calledWith = dns.calledWith_[0];\n\x1b[7m   \x1b[0m \x1b[91m                             ~~~~~~~~~~~\x1b[0m\n\n\x1b[96mtest/zone.ts\x1b[0m:\x1b[93m169\x1b[0m:\x1b[93m31\x1b[0m - \x1b[91merror\x1b[0m\x1b[90m TS2339: \x1b[0mProperty \'calledWith_\' does not exist on type \'ServiceObject<any>\'.\n\n\x1b[7m169\x1b[0m       const calledWith = zone.calledWith_[0];\n\x1b[7m   \x1b[0m \x1b[91m                              ~~~~~~~~~~~\x1b[0m\n\n\nFound 3 errors.\n\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! @google-cloud/dns@2.0.0 compile: `tsc -p .`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the @google-cloud/dns@2.0.0 compile script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /home/kbuilder/.npm/_logs/2020-05-27T12_09_02_571Z-debug.log\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! @google-cloud/dns@2.0.0 prepare: `npm run compile`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the @google-cloud/dns@2.0.0 prepare script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /home/kbuilder/.npm/_logs/2020-05-27T12_09_02_589Z-debug.log\n2020-05-27 05:09:02,603 synthtool [ERROR] > Failed executing npm install:\n\nNone\nERROR:synthtool:Failed executing npm install:\n\nNone\n2020-05-27 05:09:02,604 synthtool [DEBUG] > Wrote metadata to synth.metadata.\nDEBUG:synthtool:Wrote metadata to synth.metadata.\nTraceback (most recent call last):\n  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main\n    "__main__", mod_spec)\n  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code\n    exec(code, run_globals)\n  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>\n    main()\n  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__\n    return self.main(*args, **kwargs)\n  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main\n    rv = self.invoke(ctx)\n  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke\n    return ctx.invoke(self.callback, **ctx.params)\n  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke\n    return callback(*args, **kwargs)\n  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main\n    spec.loader.exec_module(synth_module)  # type: ignore\n  File "<frozen importlib._bootstrap_external>", line 678, in exec_module\n  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed\n  File "/home/kbuilder/.cache/synthtool/nodejs-dns/synth.py", line 13, in <module>\n    node.install()\n  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 167, in install\n    shell.run(["npm", "install"], hide_output=hide_output)\n  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run\n    raise exc\n  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run\n    encoding="utf-8",\n  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run\n    output=stdout, stderr=stderr)\nsubprocess.CalledProcessError: Command \'[\'npm\', \'install\']\' returned non-zero exit status 1.\n2020-05-27 05:09:02,651 autosynth [ERROR] > Synthesis failed\n2020-05-27 05:09:02,651 autosynth [DEBUG] > Running: git reset --hard HEAD\nHEAD is now at 68fbc61 build: do not fail builds on codecov errors (#528) (#389)\n2020-05-27 05:09:02,659 autosynth [DEBUG] > Running: git checkout autosynth\nSwitched to branch \'autosynth\'\n2020-05-27 05:09:02,664 autosynth [DEBUG] > Running: git clean -fdx\nRemoving __pycache__/\nRemoving node_modules/\nTraceback (most recent call last):\n  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main\n    "__main__", mod_spec)\n  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code\n    exec(code, run_globals)\n  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 615, in <module>\n    main()\n  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 476, in main\n    return _inner_main(temp_dir)\n  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 595, in _inner_main\n    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)\n  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 371, in synthesize_loop\n    synthesize_inner_loop(toolbox, synthesizer)\n  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 381, in synthesize_inner_loop\n    synthesizer, len(toolbox.versions) - 1\n  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch\n    synthesizer.synthesize(synth_log_path, self.environ)\n  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 119, in synthesize\n    synth_proc.check_returncode()  # Raise an exception.\n  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode\n    self.stderr)\nsubprocess.CalledProcessError: Command \'[\'/tmpfs/src/github/synthtool/env/bin/python3\', \'-m\', \'synthtool\', \'--metadata\', \'synth.metadata\', \'synth.py\', \'--\']\' returned non-zero exit status 1.\n'
```

Google internal developers can see the full log [here](http://sponge/51e18e2a-b562-4e19-987f-9aedba25285d).

--- a/synth.metadata
+++ b/synth.metadata
@@ -3,15 +3,8 @@
     {
       "git": {
         "name": ".",
-        "remote": "https://github.com/googleapis/nodejs-dns.git",
-        "sha": "c417c7fc28529772f0175a7a60958c71effdd1b0"
-      }
-    },
-    {
-      "git": {
-        "name": "synthtool",
-        "remote": "https://github.com/googleapis/synthtool.git",
-        "sha": "be74d3e532faa47eb59f1a0eaebde0860d1d8ab4"
+        "remote": "git@github.com:googleapis/nodejs-dns.git",
+        "sha": "68fbc61d8812221c6d25536a6a5e6563128b6ebb"
       }
     }
   ]

--- a/test/change.ts
+++ b/test/change.ts
@@ -73,7 +73,7 @@ describe('Change', () => {
     it('should inherit from ServiceObject', () => {
       assert(change instanceof ServiceObject);
 
-      const calledWith = change.calledWith_[0];
+      const calledWith = (change as any).calledWith_[0];
 
       assert.strictEqual(calledWith.parent, ZONE);
       assert.strictEqual(calledWith.baseUrl, '/changes');

--- a/test/index.ts
+++ b/test/index.ts
@@ -123,7 +123,7 @@ describe('DNS', () => {
     it('should inherit from Service', () => {
       assert(dns instanceof Service);
 
-      const calledWith = dns.calledWith_[0];
+      const calledWith = (dns as any).calledWith_[0];
 
       const baseUrl = 'https://dns.googleapis.com/dns/v1';
       assert.strictEqual(calledWith.baseUrl, baseUrl);

--- a/test/zone.ts
+++ b/test/zone.ts
@@ -166,7 +166,7 @@ describe('Zone', () => {
       const zone = new Zone(dnsInstance, ZONE_NAME);
       assert(zone instanceof ServiceObject);
 
-      const calledWith = zone.calledWith_[0];
+      const calledWith = (zone as any).calledWith_[0];
 
       assert.strictEqual(calledWith.parent, dnsInstance);
       assert.strictEqual(calledWith.baseUrl, '/managedZones');

